cmd/wallet-sdk-gomobile/oauth2/clientmetadata.go

```diff
@@ -0,0 +1,206 @@
+}
+
+// ResponseTypes returns the response types.
+func (c *ClientMetadata) ResponseTypes() *api.StringArray {
+	return &api.StringArray{Strings: c.goAPIClientMetadata.ResponseTypes}
+}
+
+// SetResponseTypes sets the response types.
+func (c *ClientMetadata) SetResponseTypes(responseTypes *api.StringArray) {
+	if responseTypes == nil {
+		responseTypes = api.NewStringArray()
+	}
+
+	c.goAPIClientMetadata.ResponseTypes = responseTypes.Strings
+}
+
+// ClientName returns the client name.
+func (c *ClientMetadata) ClientName() string {
+	return c.goAPIClientMetadata.ClientName
+}
+
+// SetClientName sets the client name.
+func (c *ClientMetadata) SetClientName(clientName string) {
+	c.goAPIClientMetadata.ClientName = clientName
+}
+
+// ClientURI returns the client URI.
+func (c *ClientMetadata) ClientURI() string {
+	return c.goAPIClientMetadata.ClientURI
+}
+
+// SetClientURI sets the client URI.
+func (c *ClientMetadata) SetClientURI(clientURI string) {
+	c.goAPIClientMetadata.ClientURI = clientURI
+}
+
+// LogoURI returns the logo URI.
+func (c *ClientMetadata) LogoURI() string {
+	return c.goAPIClientMetadata.LogoURI
+}
+
+// SetLogoURI sets the logo URI.
+func (c *ClientMetadata) SetLogoURI(logoURI string) {
+	c.goAPIClientMetadata.LogoURI = logoURI
+}
+
+// Scope returns the scope.
+func (c *ClientMetadata) Scope() string {
+	return c.goAPIClientMetadata.Scope
+}
+
+// SetScope sets the scope types.
+func (c *ClientMetadata) SetScope(scope string) {
+	c.goAPIClientMetadata.Scope = scope
+}
+
+// Contacts returns the contacts.
+func (c *ClientMetadata) Contacts() *api.StringArray {
+	return &api.StringArray{Strings: c.goAPIClientMetadata.Contacts}
+}
+
+// SetContacts sets the contacts.
+func (c *ClientMetadata) SetContacts(contacts *api.StringArray) {
+	if contacts == nil {
+		contacts = api.NewStringArray()
+	}
+
+	c.goAPIClientMetadata.Contacts = contacts.Strings
+}
+
+// TOSURI returns the TOS (terms of service) document URI.
+func (c *ClientMetadata) TOSURI() string {
+	return c.goAPIClientMetadata.TOSURI
+}
+
+// SetTOSURI sets the TOS (terms of service) document URI.
+func (c *ClientMetadata) SetTOSURI(tosURI string) {
+	c.goAPIClientMetadata.TOSURI = tosURI
+}
+
+// PolicyURI returns the privacy policy document URI.
+func (c *ClientMetadata) PolicyURI() string {
+	return c.goAPIClientMetadata.PolicyURI
+}
+
+// SetPolicyURI sets the privacy policy document URI.
+func (c *ClientMetadata) SetPolicyURI(policyURI string) {
+	c.goAPIClientMetadata.PolicyURI = policyURI
+}
+
+// JWKSURI returns the JWK Set document URI.
+func (c *ClientMetadata) JWKSURI() string {
+	return c.goAPIClientMetadata.JWKSURI
+}
+
+// SetJWKSURI sets the JWK Set document URI.
+func (c *ClientMetadata) SetJWKSURI(jwksURI string) {
+	c.goAPIClientMetadata.JWKSURI = jwksURI
+}
+
+// JWKS returns the JWK Set document value. If the client metadata doesn't have one, then nil is returned instead.
+func (c *ClientMetadata) JWKS() *api.JSONWebKeySet {
```

Comment:
Maybe better to rename it to JWKSet. So it is more clear that is a set of JWKs

Reply:
Done!

Reply:
we still have JWKS function, maybe better to rename it to JWKSet?

Reply:
Oops, my mistake. I thought I had changed them. They should be changed now.